### PR TITLE
MYST 355: limit openvpn client reconnects

### PR DESCRIPTION
**Pull request by zolia**
currently, if node identity changes, client auth simply fails and client hangs in reconnecting state. this PR ensures, that if auth failed, do not attempt to reconnect. Myst client now reports "not connected".

### CODE REVIEW

File: bin/server_package/config/ca.crt [hunk 1]
```diff
@@ -1,19 +0,0 @@
------BEGIN CERTIFICATE-----
```
Comment (donce) [inline bin/server_package/config/ca.crt]: Why are these removed?

Comment (zolia) [reply bin/server_package/config/ca.crt]: these are generated now, leftovers from the old auth model.

File: location/detector_test.go [hunk 2]
```diff
@@ -16,6 +16,7 @@ func TestDetectorDetectCountry(t *testing.T) {
 		{"95.85.39.36", "NL", ""},
 		{"127.0.0.1", "", ""},
 		{"8.8.8.8.8", "", "failed to parse IP"},
+		{"185.243.112.225", "", ""},
```
Comment (donce) [inline location/detector_test.go]: If we return error when we are unable to found country in database, using country detector would be much easier - if error was not returned, that means country was returned :)

File: openvpn/config.go [hunk 3]
```diff
@@ -60,6 +60,13 @@ func (c *Config) SetTLSCrypt(cryptFile string) {
 	c.AddOptions(OptionFile("tls-crypt", cryptFile))
 }
 
+func (c *Config) SetReconnectLimits() {
+	c.setParam("connect-retry-max", "2")
+	c.setParam("remap-usr1", "SIGTERM")
+	c.setFlag("single-session")
+	c.setFlag("tls-exit")
```
Comment (donce) [inline openvpn/config.go]: Some of params doesn't seem to be used for reconnection limit, which is the purpose of this method. I.e. `tls-exit` and `remap-usr1` seems to be used for allowing to exit vpn in some graceful way.

File: openvpn/factory.go [hunk 4]
```diff
@@ -40,6 +40,7 @@ func NewClientConfig(
 	config.SetClientMode(remote, 1194)
 	config.SetTLSCACertificate(caCertPath)
 	config.SetTLSCrypt(tlsCryptKeyPath)
+	config.SetReconnectLimits()
```
Comment (donce) [inline openvpn/factory.go]: If number `2` were a parameter to this method, we could see what that limit is in here - also, that would allow us to change it in factory without changing config class itself.

File: openvpn/config.go (hunk 3 above)
Comment (zolia) [reply openvpn/config.go]: i use limit in broader sense, not a number, but limitation, not to reconnect.

Comment (zolia) [reply openvpn/config.go]: ant these are parameters that enforces that limitation..

Comment (donce) [reply openvpn/config.go]: > i use limit in broader sense, not a number, but limitation, not to reconnect.

I don't quite understand - anyways, documentation for this method is missing - maybe documenting this would explain what broader sense you have?

File: openvpn/factory.go (hunk 4 above)
Comment (shroomist) [reply openvpn/factory.go]: intuitively I want to pass parameters to a function that has a `set` prefix.
perhaps `RestrictConnectRetryMax(2)`

File: bin/server_package/config/ca.crt (hunk 1 above)
Comment (donce) [reply bin/server_package/config/ca.crt]: If it was in a separate commit with an explanatory message, it would cause less confusion :)

File: location/detector_test.go (hunk 2 above)
Comment (Waldz) [reply location/detector_test.go]: return errors.New("unkown country for IP: x.x.x")

File: openvpn/config.go [hunk 5]
```diff
@@ -60,6 +60,13 @@ func (c *Config) SetTLSCrypt(cryptFile string) {
 	c.AddOptions(OptionFile("tls-crypt", cryptFile))
 }
 
+func (c *Config) SetReconnectLimits() {
+	c.setParam("connect-retry-max", "2")
```
[the remark this answers is not in the document]
Comment (zolia) [reply openvpn/config.go]: I believe no. At least not in here. This is just openvpn related. True connect / reconnect attempts should go from tequila api. Failure to connect should probably mean either real connectivity issues or failed auth due to state / outdated auth procedure in dialog stage.

File: openvpn/factory.go (hunk 4 above)
Comment (zolia) [reply openvpn/factory.go]: Restrict prob better term here. Will rename.
